### PR TITLE
Use a ChoiceBox for water shader

### PR DESCRIPTION
**Pull request by Peregrine05**
Removed the `Still water` and `Procedural water` checkboxes and replaced them with a `Water shader` dropdown that contins options: `SIMPLEX` (default), `LEGACY`, and `STILL`. Also updated the persistent settings to support the new GUI. Scenes from older versions should have their water settings preserved when loaded.

**New:**
image
image
image

### CODE REVIEW

File: chunky/src/java/se/llbit/chunky/renderer/WaterShadingStrategy.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with Chunky.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.llbit.chunky.renderer;
+
+import se.llbit.util.Registerable;
+
+public enum WaterShadingStrategy implements Registerable {
+    SIMPLEX("Simplex", "Uses configurable noise to shade the water, which prevents tiling at great distances."),
+    LEGACY("Legacy", "Uses a tiled normal map to shade the water"),
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/renderer/WaterShadingStrategy.java:23]: Why not call it `builtinNormalMap` then?

Comment (Peregrine05) [reply chunky/src/java/se/llbit/chunky/renderer/WaterShadingStrategy.java:23]: What is the determining factor of what case the ID should be?

Comment (Peregrine05) [reply chunky/src/java/se/llbit/chunky/renderer/WaterShadingStrategy.java:23]: I notice that some may be camel case and others may be all caps.

Comment (leMaik) [reply chunky/src/java/se/llbit/chunky/renderer/WaterShadingStrategy.java:23]: It's not about the case (didn't notice that) but about whether we consider the hard-coded normal map to be legacy.

**Review by leMaik — APPROVED** [review]
Looks good. I renamed `LEGACY` and improved the tooltip. Let me know what you think!

**Comment by Peregrine05** [comment]
The changes look good to me. :+1:

**Pull request merged**